### PR TITLE
Basic check to make sure the bulk import seperator is being used.

### PR DESCRIPTION
**Pull request by tonypartridge**
Added Import State function as to how the urls should be imported, enabled or disabled.

Pull Request for Issue #18844.

### Summary of Changes

Bulk Importer now has a separator missing warning when bulk import occurs without the seperator anywhere within the string.

Added Bulk Import Enable / Disable links on import by default.

### CODE REVIEW

File: administrator/components/com_redirect/controllers/links.php [hunk 1]
```diff
@@ -133,7 +133,15 @@ public function batch()
 			if (!empty($batch_urls_line))
 			{
 				$params = JComponentHelper::getParams('com_redirect');
-				$batch_urls[] = array_map('trim', explode($params->get('separator', '|'), $batch_urls_line));
+				$seperator  = $params->get('separator', '|');
```
Comment (Quy) [inline administrator/components/com_redirect/controllers/links.php]: Spelling `$separator`

File: administrator/components/com_redirect/controllers/links.php [hunk 2]
```diff
@@ -133,7 +133,15 @@ public function batch()
...
+
+				// Basic check to make sure the correct separator is being used
+				if (!\Joomla\String\StringHelper::strpos($batch_urls_line, $seperator)) {
```
Comment (Quy) [inline administrator/components/com_redirect/controllers/links.php]: Move `{` to its own line.

File: administrator/components/com_redirect/controllers/links.php [hunk 3]
```diff
@@ -133,7 +133,15 @@ public function batch()
...
+					$this->setMessage(JText::sprintf('COM_REDIRECT_NO_SEPERATOR_FOUND', $seperator), 'error');
+					$this->setRedirect('index.php?option=com_redirect&view=links');
+					return false;
+				}
+				$batch_urls[] = array_map('trim', explode($seperator, $batch_urls_line));
```
Comment (Quy) [inline administrator/components/com_redirect/controllers/links.php]: Add blank line above.

File: administrator/language/en-GB/en-GB.com_redirect.ini [hunk 4]
```diff
@@ -84,6 +86,7 @@ COM_REDIRECT_N_LINKS_UPDATED_1="1 link has been updated."
 COM_REDIRECT_NEW_LINK="New Link"
 COM_REDIRECT_NO_ITEM_ADDED="No links added."
 COM_REDIRECT_NO_ITEM_SELECTED="No links selected."
+COM_REDIRECT_NO_SEPERATOR_FOUND="The seperator %s was not found in your import."
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.com_redirect.ini]: Spelling `separator`

File: administrator/components/com_redirect/config.xml [hunk 5]
```diff
@@ -22,6 +22,17 @@
 			description="COM_REDIRECT_BULK_SEPARATOR_DESC"
 			default="|"
 		/>
+		<field
+				name="defaultImportState"
```
Comment (Quy) [inline administrator/components/com_redirect/config.xml]: Fix whitespace

**Comment by tonypartridge** [comment]
Thanks @Quy clearly tired lastnight :/

File: administrator/components/com_redirect/config.xml [hunk 6]
```diff
@@ -22,6 +22,17 @@
...
 			default="|"
 		/>
+		<field
+			name="defaultImportState"
+			type="radio"
+			label="COM_REDIRECT_DEFAULT_IMPORT_STATE_LABEL"
+			description="COM_REDIRECT_DEFAULT_IMPORT_STATE_DESC"
+			class="btn-group btn-group-yesno"
+			default="0"
+		>
```
Comment (Quy) [inline administrator/components/com_redirect/config.xml]: Indent

File: administrator/components/com_redirect/controllers/links.php [hunk 7]
```diff
@@ -133,7 +133,16 @@ public function batch()
...
 				$params = JComponentHelper::getParams('com_redirect');
-				$batch_urls[] = array_map('trim', explode($params->get('separator', '|'), $batch_urls_line));
+				$separator  = $params->get('separator', '|');
+
+				// Basic check to make sure the correct separator is being used
+				if (!\Joomla\String\StringHelper::strpos($batch_urls_line, $separator))
+				{
+					$this->setMessage(JText::sprintf('COM_REDIRECT_NO_SEPARATOR_FOUND', $separator), 'error');
+					$this->setRedirect('index.php?option=com_redirect&view=links');
+					return false;
```
Comment (Quy) [inline administrator/components/com_redirect/controllers/links.php]: Add blank line above.

File: administrator/components/com_redirect/controllers/links.php [hunk 8]
```diff
@@ -133,7 +133,16 @@ public function batch()
...
+				}
+				$batch_urls[] = array_map('trim', explode($separator, $batch_urls_line));
```
Comment (Quy) [inline administrator/components/com_redirect/controllers/links.php]: Add blank line above.

File: administrator/language/en-GB/en-GB.com_redirect.ini [hunk 9]
```diff
@@ -84,6 +86,7 @@ COM_REDIRECT_N_LINKS_UPDATED_1="1 link has been updated."
 COM_REDIRECT_NEW_LINK="New Link"
 COM_REDIRECT_NO_ITEM_ADDED="No links added."
 COM_REDIRECT_NO_ITEM_SELECTED="No links selected."
+COM_REDIRECT_NO_SEPARATOR_FOUND="The seperator %s was not found in your import."
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.com_redirect.ini]: Fix `seperator`

File: administrator/components/com_redirect/config.xml (hunk 6 above)
Comment (tonypartridge) [reply administrator/components/com_redirect/config.xml]: How do you mean? the > finishes inline with the opening <?

File: administrator/components/com_redirect/controllers/links.php (hunk 8 above)
Comment (tonypartridge) [reply administrator/components/com_redirect/controllers/links.php]: doh thanks.

File: administrator/components/com_redirect/controllers/links.php (hunk 7 above)
Comment (tonypartridge) [reply administrator/components/com_redirect/controllers/links.php]: tnanks

File: administrator/language/en-GB/en-GB.com_redirect.ini (hunk 9 above)
Comment (tonypartridge) [reply administrator/language/en-GB/en-GB.com_redirect.ini]: Fixed thanks

File: administrator/components/com_redirect/config.xml (hunk 6 above)
Comment (Quy) [reply administrator/components/com_redirect/config.xml]: It should be line 14 before your change. Check the other XML files to see the format when there are `<option>`.

Comment (tonypartridge) [reply administrator/components/com_redirect/config.xml]: Thanks! That's interesting I always thought they were double indented. I must get PHPCS setup on my MBP. Thanks @Quy what would we do without you! 

Are you a real person or someones AI Bot? ;-)

File: administrator/components/com_redirect/controllers/links.php [hunk 10]
```diff
@@ -133,7 +133,18 @@ public function batch()
...
 				$params = JComponentHelper::getParams('com_redirect');
-				$batch_urls[] = array_map('trim', explode($params->get('separator', '|'), $batch_urls_line));
+				$separator  = $params->get('separator', '|');
+
+				// Basic check to make sure the correct separator is being used
+				if (!\Joomla\String\StringHelper::strpos($batch_urls_line, $separator))
+				{
+					$this->setMessage(JText::sprintf('COM_REDIRECT_NO_SEPARATOR_FOUND', $separator), 'error');
+					$this->setRedirect('index.php?option=com_redirect&view=links');
+
```
Comment (Quy) [inline administrator/components/com_redirect/controllers/links.php]: Remove tabs.

File: administrator/components/com_redirect/controllers/links.php [hunk 11]
```diff
@@ -133,7 +133,18 @@ public function batch()
...
+				// Basic check to make sure the correct separator is being used
+				if (!\Joomla\String\StringHelper::strpos($batch_urls_line, $separator))
+				{
+					$this->setMessage(JText::sprintf('COM_REDIRECT_NO_SEPARATOR_FOUND', $separator), 'error');
+					$this->setRedirect('index.php?option=com_redirect&view=links');
+					
+					return false;
+				}
+
```
Comment (Quy) [inline administrator/components/com_redirect/controllers/links.php]: Remove tabs.

File: administrator/components/com_redirect/controllers/links.php [hunk 12]
```diff
@@ -133,7 +133,18 @@ public function batch()
 			if (!empty($batch_urls_line))
 			{
 				$params = JComponentHelper::getParams('com_redirect');
-				$batch_urls[] = array_map('trim', explode($params->get('separator', '|'), $batch_urls_line));
+				$separator  = $params->get('separator', '|');
```
Comment (Quy) [inline administrator/components/com_redirect/controllers/links.php]: Align or remove the extra space before `=`.

**Comment by tonypartridge** [comment]
@Quy done thanks!

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on b2cd2e183f1a58c18d3777251ad3d9514194c7be<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19982">issues.joomla.org/tracker/joomla-cms/19982</a>.</sub>

**Comment by anuragteapot** [comment]
I have tested this item :white_check_mark: successfully on b2cd2e183f1a58c18d3777251ad3d9514194c7be<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19982">issues.joomla.org/tracker/joomla-cms/19982</a>.</sub>

**Comment by ghost** [comment]
Ready to Commit after two successful tests.

**Pull request merged**